app/cisc/lib.go

```diff
@@ -0,0 +1,172 @@
+package main
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"io/ioutil"
+	"os"
+	"sort"
+	"strings"
+
+	"golang.org/x/crypto/ssh"
+
+	"github.com/dedis/cothority/app/lib/config"
+	"github.com/dedis/cothority/log"
+	"github.com/dedis/cothority/network"
+	"github.com/dedis/cothority/services/identity"
+	"gopkg.in/codegangsta/cli.v1"
+)
+
+// loadConfig will return nil if the config-file doesn't exist. It tries to
+// load the file given in configFile.
+func loadConfig() error {
+	log.Lvl2("Loading from", configFile)
+	buf, err := ioutil.ReadFile(configFile)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return err
+	}
+	_, msg, err := network.UnmarshalRegistered(buf)
+	if err != nil {
+		return err
+	}
+	var ok bool
+	clientApp, ok = msg.(*identity.Identity)
+	if !ok {
+		return errors.New("Wrong message-type in config-file")
+	}
+	return nil
+}
+
+// Saves the clientApp in the configfile - refuses to save an empty file.
+func saveConfig() error {
+	if clientApp == nil {
+		return errors.New("Cannot save empty clientApp")
+	}
+	buf, err := network.MarshalRegisteredType(clientApp)
+	if err != nil {
+		return err
+	}
+	return ioutil.WriteFile(configFile, buf, 0660)
+}
+
+// Asserts that the clientApp exists, else fatals
+func assertCA() {
```

Comment:
So I removed all globals - but there is now code-duplication all over the place with regard to `saveConfig` that is called all over the place where one call before was enough.